File: src/easy/final_prices_with_discount.zig
const std = @import("std");

const Allocator = std.mem.Allocator;

pub fn finalPricesWithDiscount(comptime T: type, allocator: Allocator, prices: []const T) ?[]const T {
    var result = allocator.alloc(T, prices.len) catch return null;
    std.mem.copyBackwards(T, result, prices);
    for (prices, 0..) |_, i| {
        for (prices[(i + 1)..]) |price| {
            if (price <= prices[i]) {
                result[i] = prices[i] - price;
                break;
            } else {
                result[i] = prices[i];
            }
        }
    }
    return result;
}

File: src/main/java/easy/FinalPricesWithDiscount.java
package easy;

public class FinalPricesWithDiscount {
  public int[] solve(int[] prices) {
    for (int i = 0; i < prices.length; i++) {
      for (int j = i + 1; j < prices.length; j++) {
        if (prices[j] <= prices[i]) {
          prices[i] -= prices[j];
          break;
        }
      }
    }
    return prices;
  }
}
